fe/fe-core/src/main/java/com/starrocks/lake/StarOSAgent.java

```diff
@@ -216,12 +216,34 @@ public void addWorker(long backendId, String workerIpPort) {
                     LOG.info("worker {} already added in starMgr", workerId);
                 }
             }
+            tryRemovePreviousWorker(backendId);
             workerToId.put(workerIpPort, workerId);
             workerToBackend.put(workerId, backendId);
             LOG.info("add worker {} success, backendId is {}", workerId, backendId);
         }
     }
 
+    // remove previous worker with same backend id
+    private void tryRemovePreviousWorker(long backendId) {
+        long prevWorkerId = getWorkerIdByBackendIdInternal(backendId);
+        if (prevWorkerId < 0) {
+            return;
+        }
+        try {
+            client.removeWorker(serviceId, prevWorkerId);
+        } catch (StarClientException e) {
+            // TODO: fix this corner case later in star mgr
+            LOG.error("Failed to remove worker {} with backend id {}. error: {}", prevWorkerId, backendId, e.getMessage());
+        }
+        workerToBackend.remove(prevWorkerId);
+        for (Map.Entry<String, Long> entry : workerToId.entrySet()) {
+            if (entry.getValue() == prevWorkerId) {
+                workerToId.remove(entry.getKey());
```

Comment:
ConcurrentModificationException?

Reply:
this private method is already protected by outer lock

Reply:
I mean remove item when iterate map.